### PR TITLE
Cell and column type vignette

### PR DESCRIPTION
**Pull request by jennybc**
http://rpubs.com/jennybc/cell-and-column-types

Not done yet, but I'll take any feedback you wish to give, whenever. It's mostly done.

### CODE REVIEW

File: vignettes/cell-and-column-types.Rmd [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+| empty              | `logical`, but all `NA` | *you cannot request this*     |
+| boolean            | `logical`               | `"logical"`                   |
+| numeric            | `numeric`               | `"numeric"`                   |
+| datetime           | `POSIXct`               | `"date"`                      |
+| text               | `character`             | `"text"`                      |
+| *anything*         | `list`                  | `"list"`                      |
+
+Some explanation about the weird cases in the first two rows:
+
+  * If a column falls in your data rectangle, but you do not want an associated variable in the output, specify the column type `"skip"`. Internally, these cells may be visited in order to learn their location, but they are not loaded and their data is never read.
```
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: Do we want to mention somewhere that the "rectangle" is an illusion? (i.e. there are both empty cells and cells that don't exist?)

File: vignettes/cell-and-column-types.Rmd [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+  * This will create a list-column in the output, each component of which is a length one atomic vector. The type of these vectors is determined using the logic described above. This can be useful if data of truly disparate type is arranged in a column.
+
+We demonstrate the `"list"` column type using the `"clippy.xlsx"` sheet that ship with Excel. Its second column holds information about Clippy that would be really hard to store with just one type.
+
+```{r}
+(clippy <- 
+   read_excel(readxl_example("clippy.xlsx"), col_types = c("text", "list")))
+tibble::deframe(clippy)
+sapply(clippy$value, class)
```
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: 😱 😱 😱

File: vignettes/cell-and-column-types.Rmd [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+**Contamination by embedded missing or bad data of incompatible type.** Example: missing data entered as `??` in a numeric column.
+
+  * Fix: use the `na` argument of `read_excel()` to describe all possible forms for missing data. This should prevent such cells from influencing type guessing and cause them to import as `NA` of the appropriate type.
+  
+*hmm ... none of the example sheets make a great demo here*
+  
+**Contamination of the data rectangle by leading or trailing non-data rows.** Example: the sheet contains a few lines of explanatory prose before the data table begins.
+
+  * Fix: specify the target rectangle. Use `skip` and `n_max` to provide a minimum number of rows to skip and a maximum number of data rows to read, respectively. Or use the more powerful `range` argument to describe the cell rectangle in various ways. See the examples in `read_excel()` help or the Sheet Geometry vignette for more detail.
```
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: `vignette("blah")` will get automatically linked on the pkgdown site

File: vignettes/cell-and-column-types.Rmd [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  
+The `deaths.xlsx` sheet demonstrates this perfectly. Here's how it imports if we don't specify `range` as we did above:
+
+```{r}
+deaths <- read_excel(readxl_example("deaths.xlsx"))
+print(deaths, n = Inf)
+```
+
+Non-data rows above and below the main data rectangle are causing all the columns to import as column.
```
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: column -> character

File: vignettes/cell-and-column-types.Rmd [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+options(tibble.print_min = 4, tibble.print_max = 4)
+```
+
+```{r}
```
Comment (jennybc) [inline vignettes/cell-and-column-types.Rmd]: Hmm... I already used that label on the previous chunk.

File: vignettes/cell-and-column-types.Rmd [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+with the wrinkle that datetimes are a very special flavor of numeric. A cell of any particular type can always be represented as one of any higher type and, possibly, as one of lower type. When guessing, `read_excel()` keeps a running "maximum" on the cell types it has seen in any given column. Once it has visited `guess_max` rows or run out of data, this is the guessed type for that column. There is a strong current towards "text", the column type of last resort.
+
+Here's an example of column guessing with `deaths.xlsx` which ships with readxl.
+
+```{r}
+read_excel(readxl_example("deaths.xlsx"), range = cell_rows(5:15))
+```
+
+*if I included ... `lefty`? ... here, I could show logical and complete the set*
```
Comment (jennybc) [inline vignettes/cell-and-column-types.Rmd]: You are `lefty = TRUE`. I just want a boolean.

File: vignettes/cell-and-column-types.Rmd (hunk 1 above)
Comment (jennybc) [inline vignettes/cell-and-column-types.Rmd]: I can't think of anything here that I can explain more clearly by burdening them with this. But it is good to reveal this sort of info. I think there will be a place for it in the Sheet Geometry vignette.

File: vignettes/cell-and-column-types.Rmd (hunk 5 above)
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: Yeah, running the previous chunk isn't that helpful in notebook; running this chunk is.

File: vignettes/cell-and-column-types.Rmd (hunk 1 above)
Comment (hadley) [inline vignettes/cell-and-column-types.Rmd]: Misread that as "Sweet Geometry" which I think would be a better name 😛

File: vignettes/cell-and-column-types.Rmd (hunk 5 above)
Comment (jennybc) [inline vignettes/cell-and-column-types.Rmd]: Oh but I bet your point is re: package loading and special handling for `setup` chunk.

**Pull request merged**